jenkins/aws/buildPython.sh

```diff
@@ -8,101 +8,129 @@ function main() {
   # Make sure we are in the build source directory
   cd ${AUTOMATION_BUILD_SRC_DIR}
   
-  # Is this really a python based project
-  [[ ! -f requirements.txt ]] &&
-    { fatal "No requirements.txt - is this really a python base repo?"; return 1; }
+  # Determine required tasks
+  # test is always required
+  [[ ! -z "${BUILD_TASKS}" ]] && REQUIRED_TASKS="${BUILD_TASKS}" || REQUIRED_TASKS=( "build unit" )
   
-  # Set up the virtual build environment
-  venv_dir="$(getTempDir "cota_venv_XXX")"
-  PYTHON_VERSION="${AUTOMATION_PYTHON_VERSION:+ -p } ${AUTOMATION_PYTHON_VERSION}"
+  # virtual environment is needed not only for build, but for unit and swagger tasks
+  if [[ " $REQUIRED_TASKS " =~ " build " ]] || [[ " $REQUIRED_TASKS " =~ " unit " ]] || [[ " $REQUIRED_TASKS " =~ " swagger " ]]; then
```

Comment:
consider using contains function from the utility library

another option is to use arrayFromList  to get build tasks as an array